code/run_capsule.py

```diff
@@ -1,7 +1,150 @@
-""" top level run script """
+import json
+import logging
+from pathlib import Path
 
-def run():
-    """ basic run function """
-    pass
+import contraqctor.contract as data_contract
+import pynwb
+from aind_behavior_vr_foraging.data_contract import dataset
+from dateutil import parser
+from hdmf_zarr import NWBZarrIO
+from ndx_events import NdxEventsNWBFile
+from pydantic import Field
+from pydantic_settings import BaseSettings
 
-if __name__ == "__main__": run()
\ No newline at end of file
+import utils
+
+logger = logging.getLogger(__name__)
+
+
+class VRForagingSettings(BaseSettings, cli_parse_args=True):
+    """
+    Settings for VR Foraging Primary Data NWB Packaging
+    """
+
+    input_directory: Path = Field(
+        default=Path("/data/"), description="Directory where data is"
+    )
+    output_directory: Path = Field(
+        default=Path("/results/"), description="Output directory"
+    )
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
+        level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+    )
+
+    settings = VRForagingSettings()
+    primary_data_path = tuple(settings.input_directory.glob("*"))
+    if not primary_data_path:
+        raise FileNotFoundError("No primary data asset attached")
+
+    if len(primary_data_path) > 1:
+        raise ValueError(
+            "Multiple primary data assets attached. Only single asset needed"
+        )
+
+    session_json_path = tuple(settings.input_directory.glob("*/session.json"))
+    data_description_json_path = tuple(
+        settings.input_directory.glob("*/data_description.json")
+    )
+    if not session_json_path:
+        raise FileNotFoundError("Primary data asset has no session json file")
+    if not data_description_json_path:
+        raise FileNotFoundError(
+            "Primary data asset has no data description json"
+        )
+
+    with open(session_json_path[0], "r") as f:
+        session_json = json.load(f)
+    with open(data_description_json_path[0], "r") as f:
+        data_description_json = json.load(f)
+    logger.info(
+        f"Found primary data {data_description_json['name']}. \
+        Starting acquisition nwb packaging now"
+    )
+
+    # pull version from here - file always assumed to exist at that path
+    task_input_logic_path = (
+        primary_data_path[0] / "behavior" / "Logs" / "tasklogic_input.json"
+    )
+    if not task_input_logic_path.exists():
+        raise FileNotFoundError(
+            f"No task logic input file found at path {task_input_logic_path}"
+        )
+
+    with open(task_input_logic_path, "r") as f:
+        task_input_logic = json.load(f)
+
+    contract_version = task_input_logic["version"]
+    logger.info(f"Using data contract version {contract_version}")
+    vr_foraging_dataset = dataset(
+        primary_data_path[0], version=contract_version
+    )
+    exec = vr_foraging_dataset.load_all()  # load tree structure
```

Comment:
If you are not loading behavior-videos, you can do `vr_foraging["Behavior"].load_all()`